staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -545,19 +546,26 @@ func (r *Reflector) watch(ctx context.Context, w watch.Interface, resyncerrc cha
 					logger.V(2).Info("Watch returned 429 - backing off", "reflector", r.name, "type", r.typeDescription)
 					select {
 					case <-stopCh:
-						return nil
+						return w, nil
 					case <-r.backoffManager.Backoff().C():
+						w.Stop()
```

Comment:
We might be able to avoid calling this in 3 places in this function by extracting out an inner function, but that might make it harder to see what changed.